pkg/traces/instance.go

```diff
@@ -160,68 +121,70 @@ func (i *Instance) buildAndStartPipeline(ctx context.Context, cfg InstanceConfig
 		ctx = context.WithValue(ctx, contextkeys.Logs, logs)
 	}
 
-	if cfg.ServiceGraphs != nil {
-		ctx = context.WithValue(ctx, contextkeys.PrometheusRegisterer, reg)
-	}
-
 	factories, err := tracingFactories()
 	if err != nil {
 		return fmt.Errorf("failed to load tracing factories: %w", err)
 	}
-	i.factories = factories
 
 	appinfo := component.BuildInfo{
 		Command:     "agent",
 		Description: "agent",
 		Version:     build.Version,
 	}
 
-	settings := component.TelemetrySettings{
-		Logger:         i.logger,
-		TracerProvider: trace.NewNoopTracerProvider(),
-		MeterProvider:  metric.NewNoopMeterProvider(),
+	err = enableOtelFeatureGates(
+		"telemetry.useOtelForInternalMetrics",
+		"telemetry.disableHighCardinalityMetrics")
+	if err != nil {
+		return err
 	}
```

Comment:
I added one now. It's a little big because I wanted to include an example metric to illustrate it better. But if you think it's too verbose I could remove that part. Personally I think the explanation helps and is worth the extra characters :)